### DISCUSSION TITLE
Getting slow speeds with a 4090, is this expected?

### DISCUSSION THREAD

**Discussion by Nikerino** [Anjok07/ultimatevocalremovergui#1193] [Q&A] [answered]
Problem:
I started toying around with different settings and found Ensemble mode, and I'm now using that to separate vocals from instrumental, with both at the max spec.
I'm using the models:
* MDX-Net UVR_MDX-NET Main
* MDX-Net UVR_MDX-NET Inst HQ3
with Max Spec / Max Spec to separate Vocals / Instrumental.

I noticed that in this video: https://www.youtube.com/watch?v=RvZM381wdmQ - the person is using similar settings with similar models and his separation takes less than a minute.

My separation using these on a 5 minute audio is taking a fairly large time, 10+ minutes - which I found odd because I'm sporting a 4090 at the moment.

Does this time seem accurate for what I'm trying to do? Is there a chance my CPU is being used instead of my GPU? Task Manager make it look like my GPU has memory allocated and is definitely being used, but it surprises me that it would be this slow.

**Comment by LordYAKY** [comment]
did you try another song? it might be due to how complex a song is even for a 4090 (its just a theory i dont have 4090 to try), like is it as simple as solo vocals + drums + bass and another inst or it has more stuff going on that needs working.

**Reply by Nikerino** [reply]
I've tried it with How To Save A Life (The Fray) and Perfect (Ed Sheeran) and had similar results on both.
Perfect is a primarily vocal song so I'd be surprised if the complexity is too much on that one. (Admittedly though I don't really need to use Ensemble mode for that song).

**Reply by LordYAKY** [reply, marked as answer]
I see, then there is other reasons to why its happening. 
I could think of few 
- The App isn't utilizing the whole GPU and is using the CPU with it (low probability) 
- You don't have enough RAM on your MB to support the GPU (since the app uses RAM, i have some doubt for this one) 
- The person's MotherBoard (you have watched on youtube) is more optimized for the components which is why its making the GPU like Hercules  (i REALLY DOUBT this one too) 
- last one simply is that he overclocked the gpu 💀 (higher probability) 

Note : I know you have your drivers uptodate but that also another problem that can make the gpu slower in some builds. 

Hope I've said something useful ☕

-

**Reply by Nikerino** [reply]
Appreciate the reply, here's my notes on each:
1. I believe the app is utilizing only the GPU - using Task Manager I can see the VRAM allocated in my GPU, and no shared memory is being used (thus I believe it's unlikely both GPU & CPU are being used)
2. Agreed on doubting this - I have 64 GB of RAM and the GPU has 24GB of VRAM, I've played around with settings to ensure I'm only using at max 75-90% of VRAM.
3. Potentially, I don't know his specs.
4. Perhaps as well - I haven't messed around with overclocking before. Maybe I'll give it a shot.

Either way, there might not be a clear answer. Thanks for taking the time to reply.

**Reply by LordYAKY** [reply]
Happy to help, the app is open source so glitches and odd situations tend to happen 👌
